wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -708,19 +712,21 @@ def reset_index(self):
         self.get_rebuilder().reset_index()
 
     def add_type(self, model):
-        self.get_index().add_model(model)
+        self.get_index_for_model(model).add_model(model)
 
     def refresh_index(self):
         self.get_index().refresh()
 
     def add(self, obj):
-        self.get_index().add_item(obj)
+        self.get_index_for_model(type(obj)).add_item(obj)
 
     def add_bulk(self, model, obj_list):
-        self.get_index().add_items(model, obj_list)
+        self.get_index_for_model(model).add_items(model, obj_list)
 
     def delete(self, obj):
-        self.get_index().delete_item(obj)
+        self.get_index_for_model(type(obj)).delete_item(obj)
 
+# Backwards compatibility
+ElasticSearch = ElasticsearchSearchBackend
```

Comment:
> Can you give me an idea, why do we need a different messages? Can we just add a sentence to "Upgrade considerations" and say something like this in a deprecation warning?

There will be a lot of users that have an old backend name in their settings, we should point out that they need to change their ``WAGTAILSEARCH_BACKENDS`` setting.